frontend/src/components/joystick/index.js

```diff
@@ -20,71 +20,180 @@ const useStyles = makeStyles()((theme) => {
   }
 })
 
-const Joystick = observer(() => {
-  const [gamepad, setGamepad] = useState(null)
-  useGamepads(gamepads => setGamepad(gamepads[0]))
+const ButtonLabel = {
+  DPadUp: 'DPadUp',
+  DPadDown: 'DPadDown',
+  DPadLeft: 'DPadLeft',
+  DPadRight: 'DPadRight',
+  Y: 'Y',
+  X: 'X',
+  A: 'A',
+  B: 'B',
+  LB: 'LB',
+  RB: 'RB',
+  LT: 'LT',
+  RT: 'RT',
+}
 
-  useEffect(() => {
-    if (gamepad) {
-      // this effect is invoked any time when gampad button or axes pressed, so we can send event to the BE
-      console.log('gamepad', gamepad)
-    }
-  }, [gamepad])
+const buttonState = {
+  [ButtonLabel.DPadUp]: false,
+  [ButtonLabel.DPadDown]: false,
+  [ButtonLabel.DPadLeft]: false,
+  [ButtonLabel.DPadRight]: false,
+  [ButtonLabel.Y]: false,
+  [ButtonLabel.X]: false,
+  [ButtonLabel.A]: false,
+  [ButtonLabel.B]: false,
+  [ButtonLabel.LB]: false,
+  [ButtonLabel.RB]: false,
+  [ButtonLabel.LT]: false,
+  [ButtonLabel.RT]: false,
+}
+
+const AxisLabel = {
+  LeftStickX: 'LeftStickX',
+  LeftStickY: 'LeftStickY',
+  RightStickX: 'RightStickX',
+  RightStickY: 'RightStickY',
+}
+
+const axisState = {
+  [AxisLabel.LeftStickX]: 0,
+  [AxisLabel.LeftStickY]: 0,
+  [AxisLabel.RightStickX]: 0,
+  [AxisLabel.RightStickY]: 0,
+}
 
+const axisButtonL3 = 'l3'
+const axisButtonR3 = 'r3'
+
+const AxisMoveRadius = 10
+
+const Joystick = observer(() => {
+  const [buttons, setButtons] = useState(buttonState)
+  const [axis, setAxis] = useState(axisState)
+  const [axisButtonPressed, setAxisButtonPressed] = useState('')
+  const [axisShift, setAxisShift] = useState([0,0])
   const theme = useTheme()
   const { classes } = useStyles()
   
   const handleClickEvent = (event) => {
     const { target } = event
-    
-    if (target.dataset.button && event.type === "mousedown") {
-      target.classList.add('active')
+    const buttonName = target.dataset.button
+    const isMouseDownEvent = event.type === 'mousedown'
+    if (buttonName in buttons) {
+      setButtons((currentButtonsState) => ({...currentButtonsState, [buttonName]: isMouseDownEvent}))
     } else {
-      target.classList.remove('active')
+      if(buttonName === axisButtonL3 || buttonName === axisButtonR3) {
```

Comment:
reformat this file please, or maybe it's probably better setting up prettier

Reply:
done